web-component/src/AsyncApiWebComponent.tsx

```diff
@@ -15,20 +17,107 @@ export class AsyncApiWebComponent extends React.Component<
   }
 
   render() {
+    // Angular 11.0.7 running in 'production mode + enforced stricter type
+    // checking and stricter bundle budgets' mode showed requirement of explicit
+    // definition of conditions, hence visually excessive code.
+    // Both undefined and 'undefined' are used because first has typeof
+    // 'undefined' and the second has typeof 'string'. And these differences do
+    // matter and work differently in current Angular's usecase, because Angular
+    // calls 'AsyncApiWebComponent.prototype.render' multiple times for each
+    // property specified in function 'register' (while static HTML webpage
+    // calls it only once) +1 more time at the end - assigning primitive type
+    // `undefined` to undefined properties first time and assigning string
+    // `'undefined'` each next time.
+    let schemaUrl;
+    let schemaFetchOptions;
+
     const finalCssImportPath =
       this.props.cssImportPath || 'assets/asyncapi.css';
 
+    const thisPropsSchema =
```

Comment:
I think that these options like `when schemaUrl is undefined but schema.url is not, then merge schema.requestOptions or use schemaFetchOptions prop` is unnecessary and creates a lot of if-logic. I think that if `schemaUrl` exists, then we should go only with `schemaUrl` and `schemaFetchOptions` props, and don't lookup for `schema` prop. So logic should looks like:

```js
const props = this.props;
let schemaUrl = props.schemaUrl;
let schemaFetchOptions = props.schemaFetchOptions;

if (schemaUrl === undefined || schemaUrl === "undefined") {
  schemaUrl = undefined;
}
if (schemaFetchOptions === undefined || schemaFetchOptions === "undefined") {
  schemaFetchOptions = undefined;
}

let schema = props.schema || {};
if (schemaUrl) {
   const schemaRequestOptions = schemaFetchOptions
      ? JSON.parse(JSON.stringify(schemaFetchOptions))
      : {};
   schema = { url: schemaUrl, requestOptions: schemaRequestOptions };
}
```

as you can see it is simpler and doesn't require a lot of if-statements that are difficult to understand :)